code/game/machinery/teleporter.dm

```diff
@@ -168,6 +168,8 @@
 /obj/machinery/computer/teleporter/proc/set_target(var/obj/O)
 	src.locked = O
 	GLOB.destroyed_event.register(locked, src, .proc/target_lost)
+	if(hub && station && station.engaged)
+		hub.portal_visuals.setup_visuals(locked)
```

Comment:
Should probably be a proc on `/obj/machinery/teleport/hub` rather than an outsider poking at a proc on a var on a `hub`